Feature/add/region design to multi region feature

havard024 opened the pull request:
see #364 for details

havard024 committed:
Add design for region view

havard024 committed:
Present modal when clicking on a region card

- Rename RegionView to RegionCardView
- Move rounded border and shadow style out of RegionCardView so we can use it as is in RegionView
- Wrap RegionCardView in a button that opens a dummy modal

havard024 committed:
Add RegionView design

havard024 committed:
Rename Region to RegionGroup

havard024 committed:
Fix Build input file cannot be found

Fixes issue after rename RegionView to RegionGroupView using xcodes refactor -> rename feature

sgoodwin commented on OstelcoStyles/OstelcoColor.swift:
```diff
@@ -11,13 +11,15 @@ import SwiftUI
 
 /// Colors from https://app.zeplin.io/project/5c8b989f46989524fb0258ac/styleguide
 public enum OstelcoColor: String, CaseIterable {
+    case azul
```
Are these the color names florian wanted? I thought we were using use-based names rather than color names?

havard024 replied:
There was no name for this color in zeplin, so I asked zeplin to name it for me, can update the name when it comes from design into zeplin

bloomingbridges replied:
Hi guys, 
I'm still in the process of updating the Prod project. So far the only project that uses the new Component Library (and by extension has the most up-to-date styleguide with the right colours) is the OYA MultiSIM UI project.

Commit:
Merge branch 'develop' into feature/add/region-design-to-multi-region-feature

Commit:
Merge branch 'develop' into feature/add/region-design-to-multi-region-feature

havard024 committed:
Render available ESims based on location

- Add regions to AppStore
- Add AppStore through environment for CoverageView so we can use it in multiple views without passing it as a parameter
- Split out logic for rendering available ESims based on location
- Show error message if we don't have your location
- Show ESim unavailable message if current location is not part of list of regions
- Only show GET button if ESIM for given region is not installed

havard024 committed:
Merge branch 'develop' into feature/add/region-design-to-multi-region-feature

havard024 committed:
Remove hardcoded list of available ESims from CoverageView.

havard024 committed:
Render regionGroups dynamically.

havard024 committed:
Render lint of available countries and countries in region group based on configuration.

havard024 committed:
Mark country has installed if any region who gives data to country contains a sim profile

havard024 committed:
Simplify filter logic for regions and region groups.

havard024 committed:
Version Bump to 4

havard024 committed:
Pass region to RegionOnboardingCoordinator.

- Instead of calculating the region inside the RegionOnboardingCoordinator, it is now selected before and pass into RegionOnboardingCoordinator
- Create a RegionGroup for loltel

havard024 commented:
RegionOnboardingCoordinator logic is not working as expected, thus preventing users from completing adding an ESim.

Commit:
Merge branch 'develop' into feature/add/region-design-to-multi-region-feature

havard024 committed:
Fetch region groups and country code to region codes map from remote config.

- Load remote config when app opens
- Fetch new remote config values when app enters foreground
- Use remote config to populate countryCodeToRegionCodeMap and regionGroups when AppStore is initially created. That means you need to kill the app and start again to see any changes of remote config in the UI.

havard024 committed:
Merge branch 'feature/add/region-design-to-multi-region-feature' of github.com:ostelco/ostelco-ios-client into feature/add/region-design-to-multi-region-feature

havard024 committed:
Update TODO comments.
